Mk fix app strings

mkangia opened the pull request:
https://dimagi-dev.atlassian.net/browse/ICDS-225

Specific comment being addressed is [this](https://dimagi-dev.atlassian.net/browse/ICDS-225?focusedCommentId=12953&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-12953)
thank you @orangejenny for tracking this down.

fyi @Rohit25negi

mkangia committed:
do not call set on config_ids

config_ids is a list, calling set is redundant

mkangia committed:
confirm that all static report configs are found

mkangia committed:
rearrange configs in the same order of ids as received

mkangia commented:
I have tested just the method change and it appears to be working correctly now. Will test it robustly before merging though.

orangejenny commented on corehq/apps/userreports/models.py:
```diff
@@ -817,7 +817,7 @@ def by_ids(cls, config_ids):
 
         return_configs = []
-        for config_id in set(config_ids):
+        for config_id in config_ids:
```
Can't you leave the `set`? The later logic - creating the dict and populating the return value based on that - is enough to fix the bug.

mkangia replied:
having set is redundant here. Its a list that is passed. I don't see why we should make it a set.

orangejenny replied:
I'm guessing it's in case there are duplicates in the config_ids passed in, so the loop doesn't run unnecessarily.

mkangia replied:
I guess i was trying to maintain the same order of reports/configs by ids and hence didn't see the obvious. Moving the re-ordering to the calling method post discussion and hence adding this back.

orangejenny commented:
What you have certainly works, but as an alternative to think about, on the way to work this morning I was thinking about fixing this by making `by_ids` to return a dict of id => report. The function name `by_some_property` makes me expect the return value to be a dict keyed by `some_property`.

As is, you're fixing the formerly-incorrect assumption that `by_ids` will return a list in the same order as the list passed in, but I think it'd be slightly better to not have `by_ids` do anything with the ordering, and instead force the caller to deal with it, making it obvious that you can't depend on ordering by returning a dict instead of a list. There's only one call to `by_ids`, so you wouldn't need to update a bunch of places.

orangejenny commented:
(added feature flag label because this is specific to mobile UCR)

millerdev reviewed (commented):
Deferring to Jenny, who seems to have a much better grasp of the context.

mkangia committed:
return configs by ids instead as dict

call to this method should not expect cofigs to be returned in same order so better to return as a dict mapped to id instead

mkangia committed:
reorder at _load_reports

mkangia commented:
thanks @orangejenny 

> on the way to work this morning
You do that too often.. :)

i made the change to by_ids since that makes sense. The change i made in `get_report_configs` would still be needed because we are handling static + dynamic reports there and hence the re-ordering is still needed in case we have a mix up of static and dynamic report ids.

initially i did the re-ordering at _load_reports which i skipped since it was needed any more but just committed to share that as well. We can either re-order at _load_reports or `get_report_configs`. I would go for `get_report_configs` but in case you feel otherwise fine to do it at _load_reports though that would have the need to re-order in case `get_report_configs` is used with multiple ids again later.

orangejenny commented:
> that would have the need to re-order in case get_report_configs is used with multiple ids again later

I do like having the reordering in _load_reports, because it's intuitive to me to have the callee do less work, but given that this function is currently only used once and everyone is bad at predicting the future, who knows whether or not future calls will also want the returned reports to be in the same order as the config, so I don't feel strongly about which implementation you go with.

mkangia committed:
use set to avoid duplicates and skip reordering at get_report_configs

mkangia committed:
avoid calling values

mkangia commented:
Thanks for the feedback @orangejenny 

I chose to re-order at the first calling method because it could have a mixed order of dynamic and static report ids and hence it was needed there in either case.
I have updated.

orangejenny commented:
@mkangia cool, makes sense to me!
